### PR TITLE
[IMP] stock, mrp: make _split() works in batch

### PR DESCRIPTION
**Pull request by Whenrow**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Whenrow** [comment]
On a picking with 200 stock_moves, the backorder creation takes :

    Before the patch : 16.5 sec
    After the patch : 11 sec

[flamegraphs.zip](https://github.com/odoo/odoo/files/4214960/flamegraphs.zip)
Check the flamegraph attached to see the difference

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -1563,17 +1563,15 @@ def _split(self, qty, restrict_partner_id=False):
...
-        # FIXME: pim fix your crap
         # Update the original `product_qty` of the move. Use the general product's decimal
         # precision and not the move's UOM to handle case where the `quantity_done` is not
         # compatible with the move's UOM.
         new_product_qty = self.product_id.uom_id._compute_quantity(self.product_qty - qty, self.product_uom, round=False)
         new_product_qty = float_round(new_product_qty, precision_digits=self.env['decimal.precision'].precision_get('Product Unit of Measure'))
         self.with_context(do_not_unreserve=True, rounding_method='HALF-UP').write({'product_uom_qty': new_product_qty})
-        new_move = new_move._action_confirm(merge=False)
-        return new_move.id
+        return new_move_vals
```
Comment (ryv-odoo) [inline addons/stock/models/stock_move.py:1574]: Can you change the doc of the function please ?

Comment (ryv-odoo) [reply addons/stock/models/stock_move.py:1574]: `return new_move_vals[0]` and `ensure_one()`

**Review by ryv-odoo — COMMENTED** [review]
Some suggestions

File: addons/mrp/models/mrp_production.py [hunk 2]
```diff
@@ -1379,16 +1379,21 @@ def _generate_backorder_productions(self, close_mo=True):
...
                         if move.raw_material_production_id:
-                            new_move.raw_material_production_id = backorder_mo.id
+                            move_vals[0]['raw_material_production_id'] = backorder_mo.id
                         else:
-                            new_move.production_id = backorder_mo.id
-                        (move | new_move)._do_unreserve()
-                        (move | new_move)._action_assign()
+                            move_vals[0]['production_id'] = backorder_mo.id
+                        new_moves_vals.append(move_vals[0])
+                new_moves = self.env['stock.move'].create(new_moves_vals)
```
Comment (ryv-odoo) [inline addons/mrp/models/mrp_production.py:1394]: `new_moves` isn't confirm anymore.

**Comment by Whenrow** [comment]
robodoo r+ rebase ff

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**